homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -203,6 +202,14 @@ def __init__(self):
 
         _LOGGER.debug("Discovered device %s (%s - %s)", name, model, hkid)
 
+        # Device isn't paired with us or anyone else.
+        # But we have a 'complete' config entry for it - that is probably
+        # invalid. Remove it automatically.
+        existing = find_existing_host(self.hass, hkid)
+        if not paired and existing:
+            await self.hass.config_entries.async_remove(existing.entry_id)
```

Comment:
This logic was after `self._abort_if_unique_id_configured()` previously and because of that wouldn't get called like when it was added.